js/timeout_error.go

```diff
@@ -18,44 +18,41 @@
  *
  */
 
-package lib
+package js
```

Comment:
This is great since it was only used in `js`, but could we rename the file to `errors.go`? If we follow the convention that every package should return its own set of errors, naming it `timeout_error.go` is not open for expansion and someone might define errors elsewhere.

Reply:
didn't realize `InitContextError` even existed :face_with_head_bandage: :+1: what you suggest sounds good to me

Reply:
Unfortunately, we can't do a lot about `InitContextError`. It is essentially used as the equivalent of `throw new Error()`, only from the Go side. We can't un-export it, since it's used by multiple different packages. And we can't move it to the `js/` package from `js/common/`, since that causes an import loop :disappointed: 

As things currently are in this PR, the `InitContextErrors` are detected as script errors, they bubble up from goja and have an appropriate stacktrace. For example, when using a `group()` in the init context:
```
ERRO[0000] GoError: Using group() in the init context is not supported
	at go.k6.io/k6/js/common.Bind.func1 (native)
	at file:///path/to/script.js:28:19(59)  hint="error while initializing VU #2 (script exception)"
```

The biggest problem IMO is that `GoError` and `at go.k6.io/k6/js/common.Bind.func1 (native)` are very confusing, but I'll create a separate issue about that.

Reply:
Created a new issue in https://github.com/k6io/k6/issues/2049

Reply:
Ah, that's a shame. We can leave it as is then, it's not a huge problem.